### DISCUSSION TITLE
Serverless Functions 2 minutes deployed vs 600ms local

### DISCUSSION THREAD

**Discussion by JR0210** [vercel/next.js#54302] [Help] [answered]
### Summary

Hi, I've got a Next v13.4.7 app using the app router to compare rightmove listings. I'm running into an issue at the moment where the deployed version on vercel takes infinitely longer for an initial PATCH request. 600ms local vs up to 2 minutes deployed.

I've attempted running a production build locally and that also works fine/the same speed as a local dev build.

Any ideas why? Is it just due to being on a hobby tier of vercel? All of the API requests do take longer in general but this one PATCH request is an outlier for taking so much longer.

**Local:**
![image](https://github.com/vercel/next.js/assets/37868868/977264f5-7f7e-46ed-8065-32c07a554a08)

**Production:**
![image](https://github.com/vercel/next.js/assets/37868868/34b3ae13-bfca-46ea-ad18-ee1c7724696a)

API route code can be found [here](https://github.com/JR0210/property-ranker/blob/main/src/app/api/properties/route.ts). It's still a WIP and needs error handling etc.

[Deployed version](https://property-ranker.vercel.app/)

Thanks 🙂

**Comment by icyJoseph** [comment, marked as answer]
Isn't JSDOM super large?

- https://bundlephobia.com/package/jsdom@22.1.0
- https://pkg-size.dev/jsdom

So when your API is hit, it has to parse that package, and how fast it does that will depend on, how much memory the deployment has, right? Though the PATCH request doesn't necessarily touch that...

**Reply by JR0210** [reply]
Ah yeah I didn't actually clock the bundle size of that, but as you say, the PATCH request doesn't touch that library, the POST does but seems to only take about 2-3 seconds 🤔

I will look at replacing that library too just in case it causes any extra speed up, thanks

**Reply by icyJoseph** [reply]
Also, what's the payload on the PATCH request, is it large? I wonder if you are sending several MB of data, maybe that'd also contribute to poor performance.

**Reply by JR0210** [reply]
Hey, sorry been busy with work! Good ideas on the JSDOM & payload size. I'm not really sure which caused it but I've switched to node-html-parser & I've reduced the payload to only pass in the required data & the requests now vary from 78ms to 1 second. Thanks! 😄
